pkgs/os-specific/linux/ffado/default.nix

```
@@ -1,35 +1,56 @@
-{ stdenv, fetchurl, scons, pkgconfig, which, makeWrapper, python3
-, libraw1394, libconfig, libavc1394, libiec61883, libxmlxx3
+{ stdenv
+, mkDerivation
+, dbus
+, dbus_cplusplus
+, desktop-file-utils
+, fetchurl
 , glibmm
-, dbus, dbus_cplusplus
+, kernel
+, libavc1394
+, libconfig
+, libiec61883
+, libraw1394
+, libxmlxx3
+, pkgconfig
+, python3
+, scons
+, which
+, wrapQtAppsHook
 }:
 
 let
   inherit (python3.pkgs) pyqt5 dbus-python;
   python = python3.withPackages (pkgs: with pkgs; [ pyqt5 dbus-python ]);
-in stdenv.mkDerivation rec {
+in
+mkDerivation rec {
   pname = "ffado";
-  version = "2.4.1";
+  version = "2.4.2";
 
   src = fetchurl {
     url = "http://www.ffado.org/files/libffado-${version}.tgz";
-    sha256 = "0byr3kv58d1ryy60vr69fd868zlfkvl2gq9hl94dqdn485l9pq9y";
+    sha256 = "09dxy6fkfnvzk45lpr74hkqymii8a45jzlq6054f3jz65m8qvj3d";
   };
 
+  prePatch = ''
+    substituteInPlace ./support/tools/ffado-diag.in \
+      --replace /lib/modules/ "${kernel.dev}/lib/modules/" \
+      --replace 'stdout ("uname", "-r").rstrip ()' '"${kernel.modDirVersion}"'
```

This looks a bit fragile; a different kernel may be running, because of `nixos-rebuild switch` or because a different kernel was specified in NixOS configuration. This seems to be the best location to find modules:

    /run/current-system/kernel-modules/lib/modules/$(uname -r)/

So probably all you want to do is

```suggestion
      --replace /lib/modules/ "/run/current-system/kernel-modules/lib/modules/"
```

I haven't tested this, but it seems like an improvement.

---

Thanks for the suggestion @roberth. I was inclined to do it in a similar fashion, but this seems to be the convention used in other packages.

Unfortunately, the suggested change does not work in all cases either as "current-system" is not always equal to the running system:

```
(4s) 0 ✓ [06:22:41] miminar@minap50: nixos (master *%>) > 
$ ls -l /run/current-system/kernel-modules/lib/modules/
total 9
dr-xr-xr-x 4 root root 16  1. led  1970 4.19.109
(0s) 0 ✓ [06:24:06] miminar@minap50: nixos (master *%>) > 
$ uname -r
4.19.91
```

Can you think of any other reliable approach?

What about the following?
```
/run/booted-system/kernel-modules/lib/modules/$(uname -r)/
```

---

Oops, `booted-system` is what I had in mind, but not what I wrote. Good catch!

---

No worries, thanks for the suggestion. Fixed.